pkgs/top-level/static.nix

```diff
@@ -161,14 +161,11 @@ in {
   };
   mkl = super.mkl.override { enableStatic = true; };
   nix = super.nix.override { withAWS = false; };
-  # openssl 1.1 doesn't compile
-  openssl = super.openssl_1_0_2.override {
-    static = true;
-
-    # Don’t use new stdenv for openssl because it doesn’t like the
-    # --disable-shared flag
-    stdenv = super.stdenv;
-  };
+  openssl = (super.openssl_1_1.override { static = true; }).overrideAttrs (o: {
+    configureFlags = (removeUnknownConfigureFlags o.configureFlags) ++ [
+      "no-shared"
+    ];
```

Comment:
Please explain the rationale of the above with comments, otherwise others will just find unexplained magic incantations :)

Reply:
makes sense, done in a new commit!